CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+    message(FATAL_ERROR "Out of source build is required")
+endif ()
+
+set(CMAKE_CXX_STANDARD 23)
+set(CMAKE_C_STANDARD 23)
+enable_language(ASM_NASM)
+enable_language(C)
+enable_language(CXX)
+
+find_program(TAR tar)
+
+
+set(WARNINGS "-Wall -Werror=incompatible-pointer-types -Werror=return-type -Wno-address-of-packed-member -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-folding-constant")
+set(KERNEL_BUILD_FLAGS "-fexceptions -mcmodel=kernel -nostdlib -fno-stack-protector -ffreestanding -mno-mmx -mno-sse -mno-sse3 -mno-sse4 -mno-avx -mno-red-zone -msoft-float")
+
+option(ENABLE_UBSAN "Enable UBSan in the build")
+if (${ENABLE_UBSAN})
+    set(KERNEL_BUILD_FLAGS "${KERNEL_BUILD_FLAGS} -fsanitize=undefined")
+endif ()
+
+set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
+set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
+set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2 -g -DNDEBUG")
+set(CMAKE_C_FLAGS_MINSIZEREL "-Os -DNDEBUG")
+
+set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
+set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE})
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_C_FLAGS_RELWITHDEBINFO})
+set(CMAKE_CXX_FLAGS_MINSIZEREL ${CMAKE_C_FLAGS_MINSIZEREL})
+
+set(SRCROOT ${PROJECT_SOURCE_DIR})
+
+include(FetchContent)
+
+configure_file(grub.cfg.in grub.cfg)
+
+Set(FETCHCONTENT_QUIET FALSE)
+FetchContent_Declare(
+        libcxxrt
+        GIT_REPOSITORY https://github.com/libcxxrt/libcxxrt.git
+        GIT_TAG a0f7f5c139a7daf71de0de201b6c405d852b1dc1
+        GIT_SHALLOW TRUE
+        GIT_PROGRESS TRUE
+        PATCH_COMMAND git reset --hard && git apply --ignore-space-change --ignore-whitespace ${SRCROOT}/cmake/libcxxrt.patch
+)
+
+add_compile_definitions(LIBCXXRT_WEAK_LOCKS)
+
+if (${CMAKE_CROSSCOMPILING})
+    add_compile_definitions(__ELF__ __x86_64__)
+    include_directories(${SRCROOT}/kernel/src/stl ${SRCROOT}/libk/include ${SRCROOT}/kernel/include ${SRCROOT}/kernel/src ${SRCROOT}/kernel/src/arch/${HUGOS_ARCH})
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WARNINGS} ${KERNEL_BUILD_FLAGS}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARNINGS} ${KERNEL_BUILD_FLAGS}")
+    set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${WARNINGS} ${KERNEL_BUILD_FLAGS}")
+endif ()
+
+add_subdirectory(${SRCROOT}/kernel)
+add_subdirectory(${SRCROOT}/libk)
+add_subdirectory(${SRCROOT}/docs)
+
+FetchContent_MakeAvailable(libcxxrt)
+
+add_custom_command(
+        OUTPUT ${PROJECT_BINARY_DIR}/kernel.bin
+        OUTPUT ${PROJECT_BINARY_DIR}/convolution.map
+        DEPENDS hugos_core_kernel hugos_libk cxxrt-static ${SRCROOT}/kernel/src/linker.ld /Users/Eliyahu/Downloads/libunwind-15.0.5.src/build/lib/libunwind.a
+        COMMAND ${CMAKE_CXX_LINK_EXECUTABLE} --eh-frame-hdr --error-limit=0 -T ${SRCROOT}/kernel/src/linker.ld -o ${PROJECT_BINARY_DIR}/kernel.bin --whole-archive $<TARGET_FILE:hugos_core_kernel> --no-whole-archive $<TARGET_FILE:hugos_libk> $<TARGET_FILE:cxxrt-static>
+        /Users/Eliyahu/Downloads/libunwind-15.0.5.src/build/lib/libunwind.a
```

Comment:
libunwind path shouldn't be hardcoded

Reply:
Trying to get libunwind to build in the kernel seems to somehow crash clang - Created #5 and will resolve this for now with a copy of libunwind in the kernel sources